### PR TITLE
Form for testing

### PR DESCRIPTION
**Pull request by joywin2003**

### CODE REVIEW

File: src/app/actions/get-price.ts [hunk 1]
```diff
@@ -3,34 +3,40 @@
...
-        }
-        const discountPercentage = parseFloat(coupon.discountPercentage ?? "0");
+export const getPrice = async (
+  couponCode?: string,
+): Promise<{
+  basePrice: number;
+  discountAmount: number;
+  finalPrice: number;
+}> => {
+  const basePrice = 1000;
```
Comment (Vyshnav001) [inline src/app/actions/get-price.ts:19]: better to create a constants file and add all the constants there

File: src/app/layout.tsx [hunk 2]
```diff
@@ -1,25 +1,44 @@
...
-            </body>
-        </html>
-    );
+  return (
+    <html lang="en" className="dark">
+      <body className={inter.className}>
+        <NextSSRPlugin routerConfig={extractRouterConfig(ourFileRouter)} />
+        <Providers>
+          {children}{" "}
+          <Toaster
```
Comment (Vyshnav001) [inline src/app/layout.tsx:29]: can u wrap this inside the providers ??

File: src/components/payment.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import { getPrice } from "@/app/actions/get-price";
+import { useState } from "react";
+import { Input } from "./ui/input";
+import { toast } from "sonner";
+import getErrorMessage from "@/utils/getErrorMessage";
+
+export function Payment() {
+  const [coupon, setCoupon] = useState("");
+  const [pricing, setPricing] = useState({
+    basePrice: 1200,
```
Comment (Vyshnav001) [inline src/components/payment.tsx:13]: use the basePrice from the constants file

**Review by Vyshnav001 — CHANGES_REQUESTED** [review]

File: src/app/actions/get-price.ts (hunk 1 above)
Comment (joywin2003) [reply src/app/actions/get-price.ts:19]: I have added under constants, this should be alright or should i save it somewhere else?
```typescript
// src/constants/index.ts
export const basePrice = 1000;
export const initialDiscount = 0;
```

**Comment by joywin2003** [comment]
```Error: to fetch users: q [Error]: Dynamic server usage: Route /api/users couldn't be rendered statically because it used `request.url`. See more info here: https://nextjs.org/docs/messages/dynamic-server-error
    at W (/home/runner/work/website/website/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:21106)
   Generating static pages (6/13) 
    at Object.get (/home/runner/work/website/website/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:28459)
    at p (/home/runner/work/website/website/.next/server/app/api/users/route.js:1:649)
    at /home/runner/work/website/website/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:36264
    at /home/runner/work/website/website/node_modules/next/dist/server/lib/trace/tracer.js:140:36
    at NoopContextManager.with (/home/runner/work/website/website/node_modules/next/dist/compiled/@opentelemetry/api/index.js:1:7062)
    at ContextAPI.with (/home/runner/work/website/website/node_modules/next/dist/compiled/@opentelemetry/api/index.js:1:518)
    at NoopTracer.startActiveSpan (/home/runner/work/website/website/node_modules/next/dist/compiled/@opentelemetry/api/index.js:1:18093)
    at ProxyTracer.startActiveSpan (/home/runner/work/website/website/node_modules/next/dist/compiled/@opentelemetry/api/index.js:1:18854)
    at /home/runner/work/website/website/node_modules/next/dist/server/lib/trace/tracer.js:122:103 {
  description: "Route /api/users couldn't be rendered statically because it used `request.url`. See more info here: https://nextjs.org/docs/messages/dynamic-server-error",
  digest: 'DYNAMIC_SERVER_USAGE'
}```

Getting this error from your admin check in middlware.ts

**Review by Vyshnav001 — APPROVED** [review]
Rest everything looks good for me

**Comment by joywin2003** [comment]
then its better to merge it, furthure changes will be done in another PR

**Pull request merged**